x/leverage/README.md

```diff
@@ -154,52 +175,146 @@ Supply utilization of a token denomination is calculated as:
+3. Sort all `Special Asset Pairs` with assets matching parts of the user's position, starting with the highest `Special Collateral Weight`.
+4. For each special asser pair, match collateral tokens with borrowed tokens until one of the two runs out. The matched amounts satisfy `Collateral Value (A) * Special Collateral Weight (A,B) = Borrowed Value (B)` for each special asset pair `[A,B,CW]`. Subtract the collateral and borrowed tokens from the user's remaining position.
+5. Then sort the user's remaining collateral tokens by `Collateral Weight` and sort their remaining borrowed tokens by  the same.
+6. Starting with the highest collateral weight in each list, match collateral tokens with borrowed tokens until either collateral or borrowed tokens are exhausted. The matched amounts satisfy `Collateral Value (A) * Minimum Collateral Weight (A,B) = Borrowed Value (B)`
+7. If collateral tokens remain after the matching is complete, then the user has borrowed less than their borrow limit. `Borrow Limit = Borrowed Value + sum(collateral value * collateral weight)` summed over all remaining collateral tokens.
+8. If borrowed tokens remain after the matching is complete, then the user has borrowed more than their borrow limit. `Borrow Limit = Borrowed Value - sum(borrowed value)` summed over all remaining borrowed tokens.
+
+Note that the borrow limit described in step 7 is the user's ideal borrow limit, their maximum borrowed value if all additional borrowed tokens had collateral weight equal to the weight of the remaining collateral.
+When borrowing tokens with inferior `Borrow Factor`, the user's actual borrow limit will be lower.
+
+> Example Borrow Limit Calculation:
+>
+> Collateral: $20 ATOM + $20 UMEE + $40 STATOM
+> Borrowed: $50 ATOM
+>
+> Assume the following collateral weights: UMEE 0.35, ATOM 0.6, STATOM 0.5
+>
+> Also assume a special asset pair [STATOM, ATOM, 0.75] is in effect. STATOM gets a boost when borrowing ATOM.
+>
+> Starting at step 3 above, since prices are already given, we first isolate any special asset pairs.
+>
+> $40 STATOM with a special collateral weight of 0.75 can borrow $30 ATOM. These amounts are matched with each other.
+>
+> The user's remaining position now looks like the following (sorted by collateral weights):
+>
+> Collateral: $20 ATOM (0.6) + $20 UMEE (0.35)
+> Borrowed: $20 ATOM (0.6)
+>
+> Then collateral is matched with borrowed assets starting with the highest weights.
+>
+> First, $20 ATOM collateral is matched with $12 borrowed ATOM.
+>
+> Then, $20 UMEE collateral is matched with $7 borrowed ATOM.
+>
+> Collateral is now exhausted, and $1 borrowed ATOM remains.
+>
+> Thus the user is $1 above their borrow limit. Their borrow limit must be $50 - $1 = $49.
```

Comment:
I have it as quoted text since that is a means of distinguising it from the steps of the algorithm, or other technical details. Any for of test box which is easily distinguished when scrolling rapidly (and which is not already used for something else in the readme) would be sufficient.

Any better alternatives?